cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -99,7 +99,7 @@ using internal::OutcomeToStatus;
 using internal::ToAwsString;
 using internal::ToURLEncodedAwsString;
 
-const char* kS3DefaultRegion = "us-east-1";
+const char* kS3DefaultRegion = "";
```

Comment:
Does this need to be a special constant anymore? We don't generally do that for other parameters. Also, it's misleading since in practice the default is still us-east-1, it just comes from the SDK (right?).

Reply:
Right. We could indeed just remove it.